pkg/datamover/testutils.go

```diff
@@ -0,0 +1,438 @@
+							Value: testAwsRegion,
+						},
+						{
+							Name:  "LOCATION_ENDPOINT",
+							Value: testAwsLocationEndpoint,
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "tls-certs",
+							MountPath: "/tmp/tls/",
+							ReadOnly:  true,
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "tls-certs",
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: secret.GetName(),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	podCreated, err := cli.CoreV1().Pods(namespace).Create(ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return podCreated, nil
+}
+
+func createRepositoryServerTestService(ctx context.Context, cli kubernetes.Interface, namespace string) (*corev1.Service, error) {
+	name := fmt.Sprintf("%s%s", repositoryServerTestService, rand.String(5))
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"name": name},
+		},
+		Spec: corev1.ServiceSpec{
+			Type:     corev1.ServiceTypeNodePort,
+			Selector: map[string]string{"app": "test-repo-server"},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "test-repo-server-service-port",
+					Protocol:   "TCP",
+					Port:       51515,
+					NodePort:   31325,
+					TargetPort: intstr.FromInt(51515),
+				},
+			},
+		},
+	}
+
+	serviceCreated, err := cli.CoreV1().Services(namespace).Create(ctx, service, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return serviceCreated, nil
+}
+
+func createTestS3CredsLocationSecrets(ctx context.Context, cli kubernetes.Interface, namespace string) (*corev1.Secret, *corev1.Secret, error) {
+	key := testAwsAccessKeyId
+	val := testAwsAccessSecretKey
+	s3Creds := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-s3-creds-",
+		},
+		Type: "secrets.kanister.io/aws",
+		Data: map[string][]byte{
+			"aws_access_key_id":     []byte(key),
+			"aws_secret_access_key": []byte(val),
+		},
+	}
+	s3Location := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-s3-location-",
+		},
+		Data: map[string][]byte{
+			"type":     []byte(crv1alpha1.LocationTypeS3Compliant),
+			"bucket":   []byte(testAwsS3BucketName),
+			"path":     []byte(defaultKopiaRepositoryPath),
+			"region":   []byte(testAwsRegion),
+			"endpoint": []byte(testAwsLocationEndpoint),
+		},
+	}
+
+	s3CredsCreated, err := cli.CoreV1().Secrets(namespace).Create(ctx, s3Creds, metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	s3LocationCreated, err := cli.CoreV1().Secrets(namespace).Create(ctx, s3Location, metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return s3CredsCreated, s3LocationCreated, nil
+}
+
+func createTestKopiaTLSCertificateSecret(ctx context.Context, cli kubernetes.Interface, namespace string) (*corev1.Secret, error) {
```

Comment:
Ditto, can we reuse the utilities in `pkg/testutils`

Reply:
Done !